src/mqueryfront/src/QueryTimer.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from "react";
+
+class QueryTimer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { currentTime: 0, countDownTime: undefined, firstCountDown: true };
+    }
+
+    tick() {
+        this.setState({
+            currentTime: Math.floor(Date.now() / 1000),
+        });
+    }
+
+    componentDidMount() {
+        this.interval = setInterval(() => this.tick(), 1000);
+
+        if (this.props.eta) {
+            this.interwalCountDown = setInterval(
+                () => this.setCountDownTime(),
+                1000
+            );
+        }
+    }
+    componentWillUnmount() {
+        clearInterval(this.interval);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.qhash !== this.props.qhash) {
+            this.setState({
+                currentTime: 0,
+                countDownTime: undefined,
+                firstCountDown: true,
+            });
+        }
+    }
+
+    setCountDownTime() {
+        let finishedStatuses = ["done", "cancelled", "failed", "expired"];
```

Comment:
duplicated code with `QueryResultsStatus.js` and this file, probably move to some common file?